iothub_client/src/iothubtransport_amqp_common.c

```diff
@@ -2406,6 +2406,11 @@ void IoTHubTransport_AMQP_Common_Unregister(IOTHUB_DEVICE_HANDLE deviceHandle)
             }
             else
             {
+                // bug:8781049
+                // stop the device and reset the pending messages before destorying the device instance
+                (void)amqp_device_stop(registered_device->device_handle);
+                prepare_for_connection_retry(registered_device->transport_instance);
```

Comment:
This shouldn't be needed. Why reconnect everyone if just one device is being unregistered?

Reply:
I backed this change out from the PR. We need to address this later.